Sources/SWBUtil/FSProxy.swift

```diff
@@ -737,7 +737,11 @@ class LocalFS: FSProxy, @unchecked Sendable {
 
         // FIXME: This enumerator has unclear error handling. Does nextObject() return nil on error? That's wrong.
         return try enumerator.compactMap { next in
-            let nextPath = path.join(next as? String)
+            // FileManager.DirectoryEnumerator produces Strings that are backed
```

Comment:
Is it any different if we use the URL API instead?

Reply:
@saagarjha I'm good to land as is and track this as a future issue for someone to explore, but want to run by you to see if you agree or want to look at this?

Reply:
It's up to you. I did some basic tests and it seems like this API might actually do the conversions for us but I need to investigate it further.

Reply:
Ok, updated.

Reply:
Thanks! Resolving this one. One last open here: https://github.com/swiftlang/swift-build/pull/72/files#r1966542537